### ISSUE TITLE
Moving / Backup Azuracast in traditional installation

### ISSUE THREAD

**Issue by tonimad** [AzuraCast/AzuraCast#1471]
Hi everybody!

I have reviewed the documentation about how to make a backup of Azuracast in Docker. However, my installation is traditional. I've not found any documentation to make a backup in traditional installations.

I need to move my complete installation, stations, database, audio files, etc ... to a new server.

How can I do this? Is it possible for someone to write a small tutorial, or add this to the documentation, for the traditional installation? Thanks!

**Comment by tonimad** [comment]
Hi,

Any news about this topic?

**Comment by tonimad** [comment]
Hi @SlvrEagle23 @Vaalyn 
You have some step by step available about this?

**Comment by SlvrEagle23** [comment]
@tonimad We do not currently have an official script to handle backup/restoration for Ansible (traditional/"bare-metal") installations. In a majority of cases, it's highly recommended that you enable backups of your VPS/server at a higher level (backing up the entire contents of the disk, for example), which would likely be provided by your VPS provider as an optional add-on.

**Comment by tonimad** [comment]
@SlvrEagle23 Thanks for your answer.

My provider only allows to backup files via ftp to another server. This does not make a backup of the database, for example.

Ok...

**Alternatives**
-------------------------------

`Alternative #1`
I would like to try to do a tutorial about this, and if the tutorial is effective, contribute with it.

In general, making a backup of a complete bare-metal installation entails following these steps, is correct?:

**Making the backup**
1. Make a backup of /var/azuracast/stations (full folder)
2. Make a backup of /var/azuracast/www_tmp (full folder)
3. Make a backup of /var/azuracast/www/config/settings.php
4. Make a full backup of database

**Restore the installation to another server**
Make a fresh installation of Azuracast on the new server
Upload / replace /var/azuracast/stations (full folder)
Upload / replace /var/azuracast/www_tmp (full folder)
Upload / replace /var/azuracast/www/config/settings.php
Replace the new database with the previous database

Surely this is more difficult than I am exposing, but this can be a starting point.


`Alternative #2`
It is possible that there is an application through the command line that can perform a full backup, including the database. And then restore everything on another server. Does anyone know any application of this type?


`Alternativa #3`
You can create a task in bountysource to make a script (or similar system) that can be easy to use, to make a full backup and move a complete installation to a new server.

**Comment by Vaalyn** [comment]
@tonimad A tutorial that others could follow would be a really helpful contribution to the project.

Here some thoughts from me about `Alternative #1`:

- Backup env file
- Backup InfluxDB
- Backup Redis (Optional?)

To backup files / directories the easiest way would be to just tar.gz them, either into separate archives or into a single one.

For backing up the MariaDB it would be nice to have a small [`mysqldump`](https://mariadb.com/kb/en/library/mysqldump/) script, we could also use this to improve the backup process for Docker installations.

**Comment by tonimad** [comment]
@Vaalyn Perfect!

Let me know some doubts, before starting.

1. Backup env file:
What files do you mean? Everything in the folder /var/azuracast or to others specifics files?

2. Backup InfluxDB:
How many databases exist in the Azuracast instance? In effect, I will use mysqldump.

3. In which file is located the password of the database? And... is "root" the user?

With this information, I will do some tests manually. I will write a step by step. And this will be the tutorial.

In a second phase, I will try to make an automatic script, which can be executed with a cron.

And if everything is correct, it may be possible to automate everything to place the backups in third-party services, such as Backblaze.

4. Backup Redis:
I do not know this software ... is it decisive?

**Comment by Vaalyn** [comment]
> 1. Backup env file:
>    What files do you mean? Everything in the folder /var/azuracast or to others specifics files?

AzuraCast uses a .env file for specific configurations, [here](https://github.com/AzuraCast/AzuraCast/blob/master/azuracast.sample.env) is a link to the sample file. There should also be one with similar content in your installation.

> 2. Backup InfluxDB:
>    How many databases exist in the Azuracast instance? In effect, I will use mysqldump.

There are 3 main DB like services in the AzuraCast stack, MariaDB, InfluxDB and Redis. MariaDB is the main database and as such the most important thing for you to backup. InfluxDB is used for saving the timeseries data which is used to generate reports and graphs. Redis is the applications external cache, we use it for example to cache information about the media files for faster access to that information.

> 3. In which file is located the password of the database? And... is "root" the user?
> 
> With this information, I will do some tests manually. I will write a step by step. And this will be the tutorial.

This should be located in the .env file I mentioned earlier.

> 1. Backup Redis:
>    I do not know this software ... is it decisive?

As it is just a cache it probably won't matter that much if it's not backed up and should get regenerated when you restore from a backup and let the sync processes run.

**Comment by tonimad** [comment]
Hi @Vaalyn @SlvrEagle23

Just a little note about this topic. Currently, I have a high workload and this is the reason why I did not do anything, yet.

In a few days I will be working on this.

**Comment by SlvrEagle23** [comment]
@tonimad Not to worry; we have an experimental branch we'll be merging shortly that allows live zero-downtime backups from both Ansible and Docker installation types, and even allows migration between the two.

**Comment by tonimad** [comment]
@SlvrEagle23 Really? Great news!

Ok, let me announce that I'm talking to a streaming provider in Europe. They are interested in integrating Azuracast in their services. And my proposal with them is to pay a monthly license directly to you.

As you know, I'm an enthusiast of your project and I know that being a constant source of benefits is important. So I'll be informing you about this very soon.

**Comment by SlvrEagle23** [comment]
@tonimad We've actually just started rolling out a new resource page for resellers, outlining specifically what our license allows and how to support us in a way that allows for priority support and promotion of the reseller on our homepage: https://www.azuracast.com/resellers.html

**Comment by SlvrEagle23** [comment]
@tonimad Now that we've done some testing and refinement, we can more confidently recommend using our Backup tool, both for keeping backups on both Ansible and Docker installations, but for migrating from Ansible to Docker via:

```
cd /var/azuracast/www
./update.sh
bash docker-migrate.sh
```

Closing as this is now resolved. :)
